### PR TITLE
Fixes #9654: Allow model_fn being a member function of a class

### LINKED ISSUE #9654: Estimator's argument checking overzealous : model_fn has following not expected args: ['self']
I'm using TensorFlow (1.1) high-level API Estimators to create my neural net. But I'm using it into a class and I have to call an instance of my class to generate the model of the neural network. (Here `self.a`)

```
class NeuralNetwork(object):
  def __init__(self):
    """ Create neural net """
    regressor = tf.estimator.Estimator(model_fn=self.my_model_fn,
                                       model_dir="/tmp/data")
    // ...

  def my_model_fn(self, features, labels, mode):
  """ Generate neural net model """
    self.a = a
    predictions = ...
    loss = ...
    train_op = ...
    return tf.estimator.EstimatorSpec(
      mode=mode,
      predictions=predictions,
      loss=loss,
      train_op=train_op)
```

But I get the error : ValueError: model_fn [...] has following not expected args: ['self']. I tried to remove the self for the args of my model but got another error TypeError: … got multiple values for keyword argument. 
Solution for now (as it was suggested on [StackOverflow](http://stackoverflow.com/questions/43755609/tensorflow-estimator-model-fn-has-following-not-expected-args-self)) is to use a lambda function to wrap my function `my_model_fn` (see below) but it will be nicer without it.
```
class NeuralNetwork(object):
  def __init__(self):
    """ Create neural net """
    regressor = tf.estimator.Estimator(
        model_fn=lambda features, labels, mode: self.my_model_fn(features, labels, mode),
        model_dir="/tmp/data")
    // ...
```

### PR DESCRIPTION
**Pull request by terrytangyuan**
Fixes #9654

### CODE REVIEW

**Comment by terrytangyuan** [comment]
@tensorflow-jenkins Test this please

**Comment by terrytangyuan** [comment]
Should be fine. Classic Jenkins issue: `ERROR: Couldn't find any revision to build. Verify the repository and branch configuration for this job.` in `cpu-cmake` test.

File: tensorflow/python/estimator/estimator.py [hunk 1]
```diff
@@ -750,6 +750,8 @@ def _verify_model_fn_args(model_fn, params):
     logging.warning('Estimator\'s model_fn (%s) includes params '
                     'argument, but params are not passed to Estimator.',
                     model_fn)
+  if tf_inspect.ismethod(model_fn):
+    _VALID_MODEL_FN_ARGS.add("self")
```
Comment (ispirmustafa) [inline tensorflow/python/estimator/estimator.py]: instead of editing _VALID_MODEL..., please remove 'self' from 'args'.
also s/"/'

**Review by ispirmustafa — COMMENTED** [review]
Thank you!

**Comment by martinwicke** [comment]
CI is offline. Hopefully back soon, but no point trying to run tests right now.

**Comment by terrytangyuan** [comment]
Thanks. Fixed! @ispirmustafa @martinwicke

**Review by ispirmustafa — APPROVED** [review]

**Comment by terrytangyuan** [comment]
@tensorflow-jenkins Test this please

**Pull request merged**